HicServices/RDMP
# bugfix/RDMP 31 extraction delect ui issue

- Typo and syntax fixes
- syntax fixes
- Remove CheckForVersionMismatches, no longer needed
- Update DocumentationCrossExaminationTest.cs
- Update DatabaseTests.cs
- Fix flatfileattacher circularity
- Test typo fixes
- Syntax cleanups
- Create directory.build.props
- Fix plugin version check code
- Better plugin load failure debugging
- Better plugin load error reporting
- Plugin work
- Update ExecuteCommandPrunePlugin.cs
- Update ExecuteCommandPrunePlugin.cs
- Update Startup.cs
- Update LoadModuleAssembly.cs
- Syntax tidy, simplify PersistStringHelper
- Syntax, duplicate dictionary lookups
- Booleans and args.Empty
- Fix infinite loop in time window checker, LINQ cleanup
- Tidy up ObjectConstructor and usage
- Syntax, pattern matching
- Compile fix, remove redundancies
- Syntax cleanups
- Final syntax cleanup
- Redundancies, unnecessary generics
- Build and syntax fixes
- Tune ColumnForbidder
- Fix Microsoft.NET.Test.Sdk versioning
- System.Resources.Extensions version mismatch
- API and naming fixups
- Update DatabaseTests.cs
- ANOTable recursion fix
- Update DatabaseTests.cs
- Update DatabaseTests.cs
- Update UnitTests.cs
- Adjust docs, date bound checking, validation serialisation
- Cleanup, better logging in startup reflection
- RC4
- String interp
- Syntax, pattern matching
- Make IgnoreAllErrorsCheckNotifier singleton, fix MEF event issue
- Update IgnoreAllErrorsCheckNotifier.cs
- Syntax clean
- Fixup ThrowImmediatelyCheckNotifier.Quiet()
- Remove unit test for obsolete method
- Refactor MEF
- Update ConsoleInputManager.cs
- Syntax, MEF
- Syntax
- Update ProposeExecutionWhenTargetIsLoadStageNode.cs
- Type checking fix
- MEF caching, distincting
- Syntax clean
- Regexes, .Net 7
- API fixup for .Net 7 internal changes
- Prune all non-DLLs in plugins
- Syntax cleaning
- Cleanup
- Pattern matching
- Coalesce
- Update ExecuteCommandPrunePlugin.cs
- Cleanup, dependency update
- Cleanup
- MEF adjustments
- Fix Lazy MEF init
- Update QueryCachingCrossServerTests.cs
- Compound assignments
- Syntax
- Dictionary
- Null propogation
- Null coalesce
- Regexes
- Docs fixup
- Syntax cleanup, remove old DWM check (Win7 only)
- Tidy XML handling
- Whitespace cleanup
- Update DbDataCommandDataFlowSource.cs
- Update Packages.md
- Update RDMPConcept.cs
- Update DataLoadInfo.cs
- Dispose of MEF references everywhere
- Update WebFileDownloader.cs
- Update ResearchDataManagementPlatform.csproj
- Post merge cleanup - duplicate lines, statics, inlining variables
- Add Equ, remote old Xam settings plugin
- Update LogManager.cs
- Specify string comparison locale
- Update Startup.cs
- Update LoadModuleAssembly.cs
- Update IPluginManager.cs
- Static methods where applicable
- Remove SetupMEF, no longer needed
- More readonly fields
- Whitespace and typo fixes
- Update UsefulStuff.cs
- Statics, readonly
- Format fixups
- Pattern matching
- Whitespace, sealed classes
- Format, auto cleanup
- Docs fixup; allow non-public constructors for reflection-invoked classes
- Static and string cleanup
- Typos and string diff fixups
- Update DataFlowPipelineContext.cs
- Make ProgressLog timeout 1 hour - insane for single-row inserts, but 2 minutes is still too short...
- Update plugin pack to install to right place
- Move Terminal.Gui to Core rather than duplicating both CLI and GUI refs
- Implement async writer thread for ProgressLog
- Flush log queue unconditionally, even on duplicate DataLoadInfo closes.
- Bugfix: load Win code only in GUI mode, instead of whenever running in WinNT
- Disused files
- Add alias for mistyped Nullity predictor
- Stray newline
- Update ValuePredictsOtherValueNullity.cs
- Format cleanup
- Tidyup
- fix extraction ui crash
